x/nft/codec.proto

```diff
@@ -32,14 +32,14 @@ message AddApprovalMsg {
     bytes address = 2;
     string action = 3;
     ApprovalOptions options = 4 [(gogoproto.nullable) = false];
-    Type t = 5;
+    string t = 5;
```

Comment:
Okay, so we don't require a Type here, but the Type message still exists, right?
As we see eg. `nft.Type_Username.String()`  in other places of the code.

Maybe just replace that protobuf enum with a simple set of golang const

Reply:
@ethanfrey the idea behind that was that that'll allow us to transfer these constants to client.
if that does not make sense - sure, we can back to constants.

Reply:
Sure, makes sense. So the docs are in code in a way.

But.... the list doesn't belong in `x/nft`.
Maybe one type defined in each subpackage `x/nft/username`...
Or the total enum defined in the app `cmd/bnsd/app`

Otherwise we have a canonical list in the base nft package that is not extensible.

I like the idea, just trying to figure out where it belongs. I don't know if it is possible to define an enum in each subpackage, then combine them somehow in the app.

Don't break your head about this... just me pushing for some purity of organization which is not urgent.

Reply:
I'll move it somewhere, sure. bnsd probably makes sense